Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc

```diff
@@ -186,6 +186,8 @@ PrimaryVertexValidation::~PrimaryVertexValidation()
 {
    // do anything here that needs to be done at desctruction time
    // (e.g. close files, deallocate resources etc.)
+  if (theTrackFilter_) delete theTrackFilter_;
```

Comment:
For future reference, it would be much better to change `theTrackFilter_` and `theTrackClusterizer_` to  be `std::unique_ptr` rather than doing explicit deletes. Also, there is no need to test for `nullptr` before calling `delete` since, internally, `delete` does the same check.

Reply:
Hi @Dr15Jones, thanks for the comment. I'll keep that in mind for the next iteration.
As I got inspiration from,

https://github.com/cms-sw/cmssw/blob/efba733c40b1f49ad9f83dc706e40349a0393c13/RecoVertex/PrimaryVertexProducer/src/PrimaryVertexProducerAlgorithm.cc#L106-L107

I think same comment applies also there: maybe worth opening an issue?

Reply:
Agreed